quartz/util/imageHelper.ts

```diff
@@ -0,0 +1,115 @@
+import satori, { FontWeight, SatoriOptions } from "satori/wasm"
+import { GlobalConfiguration } from "../cfg"
+import { JSXInternal } from "preact/src/jsx"
+
+/**
+ * Get an array of `FontOptions` (for satori) given google font names
+ * @param headerFontName name of google font used for header
+ * @param bodyFontName name of google font used for body
+ * @returns FontOptions for header and body
+ */
+export async function getSatoriFont(headerFontName: string, bodyFontName: string) {
+  const headerWeight = 700 as FontWeight
+  const bodyWeight = 400 as FontWeight
+
+  // Fetch fonts
+  const headerFont = await fetchTtf(headerFontName, headerWeight)
+  const bodyFont = await fetchTtf(bodyFontName, bodyWeight)
+
+  // Convert fonts to satori font format and return
+  const fonts: SatoriOptions["fonts"] = [
+    { name: headerFontName, data: headerFont, weight: headerWeight, style: "normal" },
+    { name: bodyFontName, data: bodyFont, weight: bodyWeight, style: "normal" },
+  ]
+  return fonts
+}
+
+/**
+ * Get the `.ttf` file of a google font
+ * @param fontName name of google font
+ * @param weight what font weight to fetch font
+ * @returns `.ttf` file of google font
+ */
+function fetchTtf(fontName: string, weight: FontWeight): Promise<ArrayBuffer> {
+  return new Promise(async (resolve, reject) => {
+    // Get css file from google fonts
+    const css = await (
+      await fetch(`https://fonts.googleapis.com/css?family=${fontName}:${weight}`)
```

Comment:
Hmm I assume we only want to support googleFonts for now?

Reply:
When starting out, I def tried to make this into a component. From what I remember, it didn't really work, as this has to modify the `<head>` tag of every single page.

As for the fonts part, I want to keep the "easy" customisation options as easy as possible. Right now, `quartz.config.ts`/`theme.typography` only supports google fonts, so this should be fine. Satori requires a special font format to be able to use `fontFamilies` in their components, this just makes it so the `header`/`body` google font from `quartz.config.ts` automatically gets pulled and injected, ready to be used for image generation.

Reply:
That sounds good; thanks for the information.

Reply:
i did think about also prefetching ttf for quartz generally so that the generated files are fully self-contained, might be worth considering some point down the line!

Reply:
I have implemented an option to enable fetching fonts to `static/fonts`